crates/client/block_production/src/lib.rs

```diff
@@ -353,8 +351,13 @@ impl<Mempool: MempoolProvider> BlockProductionTask<Mempool> {
 
```

Comment:
i think you should split the on_block_time function in two: on_block_time and close_block and on_pending_block_tick should call close_block instead of on_block_time

Reply:
did something very similar, lmk wdyt?